keystore/filesystem/server_keystore_test.go

```diff
@@ -1503,3 +1509,194 @@ func TestPoisonKeyGeneration(t *testing.T) {
+	expectedKeys := []keystore.KeyDescription{
+		{ID: "poison_key", Purpose: PurposePoisonRecordKeyPair},
+		{ID: "poison_key.pub", Purpose: PurposePoisonRecordKeyPair},
+		{ID: "poison_key_sym", Purpose: PurposePoisonRecordSymmetricKey},
+		{ID: "cossack-hmac_hmac", Purpose: PurposeSearchHMAC, ClientID: []byte(hmacEncID)},
+		{ID: "cossack_storage", Purpose: PurposeStorageClientPrivateKey, ClientID: []byte(clientID)},
+		{ID: "cossack_storage.pub", Purpose: PurposeStorageClientPublicKey, ClientID: []byte(clientID)},
+		{ID: "cossack_storage_sym", Purpose: PurposeStorageClientSymmetricKey, ClientID: []byte(clientID)},
+		{ID: "secure_log_key", Purpose: PurposeAuditLog},
+		{ID: "sich_zone_sym", Purpose: PurposeStorageZoneSymmetricKey, ZoneID: []byte(zoneID)},
+	}
+	// sort to compare consistently
+	sort.Slice(expectedKeys, func(i, j int) bool {
+		return expectedKeys[i].ID < expectedKeys[j].ID
+	})
+	return expectedKeys
+}
+
+func TestListKeysSamePaths(t *testing.T) {
+	keyStore, path, err := getKeystore()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer os.RemoveAll(path)
+
+	generateEveryKey(keyStore, t)
+
+	all, err := keyStore.ListKeys()
+	if err != nil {
+		t.Fatal(err)
+	}
+	sort.Slice(all, func(i, j int) bool {
+		return all[i].ID < all[j].ID
+	})
+
+	buff := bytes.NewBuffer([]byte{})
+
+	expectedKeys := getAllExpectedKeys()
+	printKeysTable(expectedKeys, buff)
+	expected := buff.String()
+
+	buff.Reset()
+	printKeysTable(all, buff)
+	found := buff.String()
+
+	fmt.Fprintln(os.Stderr, "=> Expected Keys")
+	fmt.Fprintln(os.Stderr, expected)
+	fmt.Fprintln(os.Stderr, "=> Actual Keys")
+	fmt.Fprintln(os.Stderr, found)
+
+	if expected != found {
+		t.Fatal("lists are different")
+	}
+}
+
+func TestListKeysDifferentPaths(t *testing.T) {
+	keyPrivateDir, err := ioutil.TempDir(os.TempDir(), "private")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(keyPrivateDir)
+
+	keyPublicDir, err := ioutil.TempDir(os.TempDir(), "public")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(keyPublicDir)
+
+	encryptor, err := keystore.NewSCellKeyEncryptor([]byte("some key"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	keyStore, err := NewFilesystemKeyStoreTwoPath(keyPrivateDir, keyPublicDir, encryptor)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	generateEveryKey(keyStore, t)
+
+	all, err := keyStore.ListKeys()
+	if err != nil {
+		t.Fatal(err)
+	}
+	sort.Slice(all, func(i, j int) bool {
+		return all[i].ID < all[j].ID
+	})
+
+	buff := bytes.NewBuffer([]byte{})
+
+	expectedKeys := getAllExpectedKeys()
+	printKeysTable(expectedKeys, buff)
+	expected := buff.String()
+
+	buff.Reset()
+	printKeysTable(all, buff)
+	found := buff.String()
+
+	fmt.Fprintln(os.Stderr, "=> Expected Keys")
+	fmt.Fprintln(os.Stderr, expected)
+	fmt.Fprintln(os.Stderr, "=> Actual Keys")
+	fmt.Fprintln(os.Stderr, found)
+
+	if expected != found {
+		t.Fatal("lists are different")
+	}
+}
+
+// copied from `acra-keys list` code
```

Comment:
it's the good reason to move this function to `keystore/utils.go` (probably) and use it in tests and acra-tools too, if we already need it in several places) Maybe in the future we will need it in one more place